addons/account/models/account_journal_dashboard.py

```diff
@@ -257,178 +261,210 @@ def _get_bar_graph_select_query(self):
-
-        is_sample_data = self.kanban_dashboard_graph and any(data.get('is_sample_data', False) for data in json.loads(self.kanban_dashboard_graph))
+            graph_data = journal.get_line_graph_datas()
+            dashboard_data[journal.id].update({
+                'number_to_check': number_to_check,
+                'to_check_balance': to_check_balance,
+                'number_to_reconcile': number_to_reconcile.get(journal.id, 0),
+                'account_balance': currency.format(bank_account_balance),
+                'has_at_least_one_statement': bool(last_statement),
+                'nb_lines_bank_account_balance': nb_lines_bank_account_balance,
+                'outstanding_pay_account_balance': currency.format(outstanding_pay_account_balance),
+                'nb_lines_outstanding_pay_account_balance': nb_lines_outstanding_pay_account_balance,
+                'last_balance': currency.format(last_statement.balance_end),
+                'bank_statements_source': journal.bank_statements_source,
+                'graph_data': graph_data,
+                'is_sample_data': graph_data and any(data.get('is_sample_data', False) for data in graph_data),
+            })
 
-        return {
-            'number_to_check': number_to_check,
-            'to_check_balance': formatLang(self.env, to_check_balance, currency_obj=currency),
-            'number_to_reconcile': number_to_reconcile,
-            'account_balance': formatLang(self.env, currency.round(bank_account_balance), currency_obj=currency),
-            'has_at_least_one_statement': has_at_least_one_statement,
-            'nb_lines_bank_account_balance': nb_lines_bank_account_balance,
-            'outstanding_pay_account_balance': formatLang(self.env, currency.round(outstanding_pay_account_balance), currency_obj=currency),
-            'nb_lines_outstanding_pay_account_balance': nb_lines_outstanding_pay_account_balance,
-            'last_balance': formatLang(self.env, currency.round(last_balance) + 0.0, currency_obj=currency),
-            'number_draft': number_draft,
-            'number_waiting': number_waiting,
-            'number_late': number_late,
-            'sum_draft': formatLang(self.env, currency.round(sum_draft) + 0.0, currency_obj=currency),
-            'sum_waiting': formatLang(self.env, currency.round(sum_waiting) + 0.0, currency_obj=currency),
-            'sum_late': formatLang(self.env, currency.round(sum_late) + 0.0, currency_obj=currency),
-            'currency_id': currency.id,
-            'bank_statements_source': self.bank_statements_source,
-            'title': title,
-            'is_sample_data': is_sample_data,
-            'company_count': len(self.env.companies)
+    def _fill_sale_purchase_dashboard_data(self, dashboard_data):
+        sale_purchase_journals = self.filtered(lambda journal: journal.type in ('sale', 'purchase'))
+        if not sale_purchase_journals:
+            return
```

Comment:
more comments: if I'm not wrong, the only reason we don't make read_groups is because the amount_residual field isn't signed, right?

Reply:
Not sure why this comment is on this line, but I put a note in the commit message regarding this